### PR TITLE
CHANGELOG: add 0.4.19

### PR DESCRIPTION
**Pull request by Stebalien**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,589 @@
...
+  to the network. (note: this feature isn't new, just an example).
+* `ipfs add --offline some_file` won't send out provider records.
+* `ipfs cat --offline Qm...` won't fetch any blocks from the network.
+* `ipfs block stat --offline Qm...` is a great way to tell if a block is locally
+  available.
+
+On to the gateway, there's a new `Gateway.NoFetch` option to configure the
+gateway to only serve locally present files. This makes it possible to run an
+IPFS node as a gateway to serve content of _your_ choosing (i.e., stuff you've
+pinned locally) but no other content.
```
Comment (magik6k) [inline CHANGELOG.md]: Might be confusing as it will also serve locally present unpinned content.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,589 @@
...
+  - enhance(cmd/verify): add goroutine count to improve verify speed ([ipfs/go-ipfs#5710](https://github.com/ipfs/go-ipfs/pull/5710))
+  - Update go-mfs and go-unixfs ([ipfs/go-ipfs#5714](https://github.com/ipfs/go-ipfs/pull/5714))
+  - fix(flag/version): flag `all` should have a higher priority ([ipfs/go-ipfs#5719](https://github.com/ipfs/go-ipfs/pull/5719))
+  - commands/p2p: use new cmds lib ([ipfs/go-ipfs#5672](https://github.com/ipfs/go-ipfs/pull/5672))
+  - commands/dht: use new cmds lib ([ipfs/go-ipfs#5671](https://github.com/ipfs/go-ipfs/pull/5671))
+  - commands/object: use new cmds ([ipfs/go-ipfs#5666](https://github.com/ipfs/go-ipfs/pull/5666))
+  - commands/files: use new cmds ([ipfs/go-ipfs#5665](https://github.com/ipfs/go-ipfs/pull/5665))
+  - cmds/env: add a config path helper ([ipfs/go-ipfs#5712](https://github.com/ipfs/go-ipfs/pull/5712))
+- github.com/ipfs/dir-index-html:
+  - show hash if given ([ipfs/dir-index-html#21](https://github.com/ipfs/dir-index-html/pull/21))
```
Comment (magik6k) [inline CHANGELOG.md]: Might be worth giving this a bigger note

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,589 @@
...
+  - expand on the api error ([ipfs/go-ipfs-cmds#138](https://github.com/ipfs/go-ipfs-cmds/pull/138))
+  - set the connection close header if we have a body to read ([ipfs/go-ipfs-cmds#116](https://github.com/ipfs/go-ipfs-cmds/pull/116))
+  - print a nicer error on timeout/cancel ([ipfs/go-ipfs-cmds#137](https://github.com/ipfs/go-ipfs-cmds/pull/137))
+  - Add link traversal option ([ipfs/go-ipfs-cmds#96](https://github.com/ipfs/go-ipfs-cmds/pull/96))
+  - Don't skip stdin test on Windows ([ipfs/go-ipfs-cmds#136](https://github.com/ipfs/go-ipfs-cmds/pull/136))
+  - MakeTypedEncoder: accept results by pointer or value ([ipfs/go-ipfs-cmds#134](https://github.com/ipfs/go-ipfs-cmds/pull/134))
+- github.com/ipfs/go-ipfs-config:
+  - Gateway.NoFetch ([ipfs/go-ipfs-config#19](https://github.com/ipfs/go-ipfs-config/pull/19))
+  - add a Clone function ([ipfs/go-ipfs-config#16](https://github.com/ipfs/go-ipfs-config/pull/16))
+  - randomports: give user ability to init ipfs using random port for swarm. ([ipfs/go-ipfs-config#17](https://github.com/ipfs/go-ipfs-config/pull/17))
```
Comment (magik6k) [inline CHANGELOG.md]: Possibly worth a bigger note

**Comment by b5** [comment]
Just saw this PR in passing. daaaaaaamnnnn 0.4.19 is looking good!

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,642 @@
...
+AutoRelay is a new service that automatically chooses a public relay when it
+detects that the go-ipfs node is behind a NAT. While relaying connections
+through a third-party node isn't the most efficient way to route around NATs,
+it's a reliable fallback.
+
+To enable `AutoRelay`, set the `Swarm.EnableAutoRelay` option in the config.
+
+AutoNAT is the service AutoRelay uses to detect if the node is behind a NAT.
+It's enabled by default when you enable `AutoRelay` so you don't have to do
+anything.
```
Comment (magik6k) [inline CHANGELOG.md]: Well, you have to enable `AutoRelay`

Comment (Stebalien) [reply CHANGELOG.md]: (reworded)

**Comment by koalalorenzo** [comment]
I can't wait to have this released 👍  Thanks guys, this will help us a lot!

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,663 @@
...
+as new chunks are fetched from the network. To test this, you can run `ipfs ls
+--stream --size=false --resolve-type=false
+/ipfs/QmXoypizjW3WknFiJnKLwHCnL72vedxjQkDDP1mXWo6uco/wiki`. You probably won't
+want to wait for that command to finish, Wikipedia has a _lot_ of entries.
+
+#### HTTP Proxy
+
+This release sees a new (experimental) feature contributed by our friends at
+[Peergos](https://peergos.org): HTTP proxy over libp2p. When enabled, the local
+gateway can act as an HTTP proxy and forward HTTP requests libp2p peers. When
```
Comment (momack2) [inline CHANGELOG.md]: to or from libp2p peers?

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,663 @@
...
+
+#### CoreAPI
+
+The work on the CoreAPI refactor ([ipfs/go-ipfs#4498][]) has progressed leaps and
+bounds this release. The CoreAPI is a comprehensive programmatic interface
+designed to allow go-ipfs be used as a daemon or a library interchangeably.
+
+As of this release, go-ipfs now has:
+
+* External interface definitions in [ipfs/interface-go-ipfs-core][].
```
Comment (momack2) [inline CHANGELOG.md]: these are all missing contents

Comment (Stebalien) [reply CHANGELOG.md]: That's short-hand for `[ipfs/interface-go-ipfs-core][ipfs/interface-go-ipfs-core]` (which refers to the link definition `[ipfs/interface-go-ipfs-core]: https://github.com/ipfs/interface-go-ipfs-core` at the end of the section.

File: CHANGELOG.md (hunk 5 above)
Comment (Stebalien) [reply CHANGELOG.md]: To, thanks!

**Comment by Stebalien** [comment]
Merging so we can cut an RC.

**Pull request merged**